src/hivpy/population.py

```diff
@@ -66,6 +75,62 @@ def _create_population_data(self):
             self.data[col.HIV_STATUS] = self.hiv_status.introduce_HIV(self.data)
             self.HIV_introduced = True
 
+    def init_variable(self, name: str, init_val, n_prev_steps=0):
+        """
+           New variable will be initialised as a collection of columns.\\
+           Column names (keys) will be (name, 0) ... (name, n_prev_steps).\\
+           Updates dictionary to keep track of number of time steps being stored for each
+           variable.
+           name: string, name of variable
+           n_prev_steps: integer, number of previous iterations of this variable which need
+           to be stored (default 0).
+        """
+        self._variable_history[name] = n_prev_steps
+        for i in range(0, n_prev_steps + 1):
+            self.data[(col.name, i)] = init_val + 1
+
+    def get_sub_pop(self, conditions):
+        index = reduce(operator.and_,
+                       (op(self.data[self.get_correct_column(var)], val) for (var, op, val) in conditions))
+        return index
+
+    def apply_vector_func(self, params, func):
+        param_cols = list(map(self.get_correct_column, params))
+        return func(*param_cols)
```

Comment:
You can also use the unpacking (`*`) directly with any iterable, without converting it to a list:
```suggestion
        param_cols = map(self.get_correct_column, params)
        return func(*param_cols)
```